cmd/provenanced/cmd/root.go

```diff
@@ -347,14 +301,3 @@ func overwriteFlagDefaults(c *cobra.Command, defaults map[string]string) {
 		overwriteFlagDefaults(c, defaults)
 	}
 }
-
-// getIAVLCacheSize If app.toml has an entry iavl-cache-size = <value>, default is 781250
-func getIAVLCacheSize(options servertypes.AppOptions) int {
-	iavlCacheSize := cast.ToInt(options.Get("iavl-cache-size"))
-	if iavlCacheSize == 0 {
-		// going with the DefaultConfig value(which is 781250),
-		// the store default value is DefaultIAVLCacheSize = 500000
-		return cast.ToInt(serverconfig.DefaultConfig().IAVLCacheSize)
-	}
-	return iavlCacheSize
-}
```

Comment:
This isn't needed because we now load the defaults into viper before trying to read the configs. So if the config doesn't exist, or if the `iavl-cache-size` entry doesn't exist, it gets the default as defined on the default config object.

I tested this locally by:
1. Added a log line in `newApp` that outputs the value of `appOpts.Get("iavl-cache-size")`.
2. Ran `make build run-config`
3. Editing the `app.toml` to remove the `iavl-cache-size` entry.
4. Ran `make run`

The log line said the value was 781250.

I then redid steps 2-4 except on 3, I changed the value to 781251 instead of removing it.
This time, the log line said the value was 781251.